### ISSUE TITLE
Parameter indentation

### ISSUE THREAD

**Issue by ubuntudroid** [pinterest/ktlint#1607]
## Expected Behavior
ktlint formats some cases of parameter indendation differently from Android Studio/IntelliJ.

Consider this code:
```kotlin
String(StringBuffer(
    "abc"
))
```

After formatting it should look like this (this is also what IntelliJ formatting does):
```kotlin
String(
    StringBuffer(
        "abc"
    )
)
```

## Observed Behavior

```kotlin
String(
StringBuffer(
    "abc"
)
)
```

Interestingly ktlint doesn't change formatting if you feed it with the output of IntelliJ formatter. 🤔 So I guess it's a bug in the formatter and not in the checker.

## Steps to Reproduce
See sample in section 1. It is the most simple pre-formatting example I could come up with 😅

## Your Environment

* Version of ktlint used: 0.48.0 (although the artifact is 0.47.0, but that seems to be correct looking at ktlint's release page)
* Relevant parts of the `.editorconfig` settings: no indendation specific rules
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): [trunk](https://github.com/trunk-io/trunk-action) 0.16.1-beta
* Version of Gradle used (if applicable): 7.4
* Operating System and version: macOS 12.5.1

**Comment by paul-dingemans** [comment]
> Version of ktlint used: 0.48.0 (although the artifact is 0.47.0, but that seems to be correct looking at ktlint's release page)

How did you determine the value `0.48.0`? Please execute `ktlint --version` to see the 'real' version number.

> See sample in section 1. It is the most simple pre-formatting example I could come up with 😅

I love simple examples. Your example is not valid kotlin code however. So I have used code below for reproduction:

```
val foo = String(StringBuffer(
"abc"
))
```

The code above is indeed formatted as:
```
val foo = String(
    StringBuffer(
        "abc"
    )
)
```
This is the way of formatting since at least version 0.37.2. This way of formatting is consistent with default IntelliJ IDEA formatting. I do not have Android Studio installed. 

What is the exact command that you execute? What are your relevant `.editorconfig` settings?

**Comment by ubuntudroid** [comment]
> How did you determine the value 0.48.0? Please execute ktlint --version to see the 'real' version number.

It is 0.47.0 - I was just referring to the official release page which as of yesterday referred to this version as 0.48.0 (but linked the 0.47.0 artifact). This seems to have been fixed as of today. Anyways, I'm using 0.47.0. 😅 

> I love simple examples. Your example is not valid kotlin code however.

Hum, why isn't this valid Kotlin code? 🤔 I mean, I'm not saying that it makes much sense to create an object and not assign it, but it is valid code and the compiler agrees. 😏 

Nevertheless, when running your slightly modified sample I also get the wrong formatting:

```kotlin
val foo = String(
StringBuffer(
    "abc"
)
)
```

I'm executing the following command:

```
ktlint -F Test.kt
```

My `.editorconfig` looks like this:

```
[*.{kt,kts}]
max_line_length = 140
indent_size = unset
# noinspection EditorConfigKeyCorrectness - used by ktlint
ktlint_disabled_rules = final-newline,max-line-length,import-ordering
```

**Comment by paul-dingemans** [comment]
> How did you determine the value 0.48.0? Please execute ktlint --version to see the 'real' version number.
> t is 0.47.0 - I was just referring to the official release page which as of yesterday referred to this version as 0.48.0 (but linked the 0.47.0 artifact). This seems to have been fixed as of today. Anyways, I'm using 0.47.0. 😅

The official release page adresses this version as `0.47.0`. Maybe you go confused by some references to version `0.48.0` in which some API changes will be introduced.

> Hum, why isn't this valid Kotlin code? 🤔 I mean, I'm not saying that it makes much sense to create an object and not assign it, but it is valid code and the compiler agrees. 😏

Exactly that. My compiler does complain about expecting a top level declaration. When you run ktlint on your original code sample, you will also see this warning:
```
Not a valid Kotlin file (1:1 expecting a top level declaration) (cannot be auto-corrected) ()
```

I still can not reproduce the problem. Can you please run with `--trace` flag added to the command and report entire log back?

**Comment by ubuntudroid** [comment]
> Exactly that. My compiler does complain about expecting a top level declaration.

My bad, I had a class declaration around the code, but didn't explicitly mention it here - sorry about the confusion! 🙏 

> I still can not reproduce the problem. Can you please run with --trace flag added to the command and report entire log back?

Sure, here you go:

<details>
<summary>Click here to expand</summary>

```
17:49:47.835 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "baseline" id.
17:49:47.836 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "checkstyle" id.
17:49:47.836 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "json" id.
17:49:47.836 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "format" id.
17:49:47.836 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "html" id.
17:49:47.836 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "plain" id.
17:49:47.836 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "sarif" id.
17:49:47.837 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Initializing "plain" reporter with {verbose=false, color=false, color_name=DARK_GRAY, format=true}
17:49:47.856 [pool-1-thread-1] TRACE com.pinterest.ktlint.internal.KtlintCommandLine - Checking <redacted>/Test.kt
17:49:48.377 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.internal.ThreadSafeEditorConfigCache - Creating new EditorConfig cache entry for path <redacted>/.editorconfig
17:49:48.378 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.internal.EditorConfigLoader - Resolving .editorconfig files for <redacted>/Test.kt file path:
        max_line_length: 140
        indent_size: unset
        ktlint_disabled_rules: final-newline,max-line-length,import-ordering
        tab_width: unset
17:49:48.388 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'indent_style' was found. Value has been defaulted to 'space'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.388 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - Value of '.editorconfig' property 'indent_size' is remapped from 'unset' to '-1'
17:49:48.393 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'indent_style' was found. Value has been defaulted to 'space'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.393 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - Value of '.editorconfig' property 'indent_size' is remapped from 'unset' to '-1'
17:49:48.402 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'ij_kotlin_packages_to_use_import_on_demand' was found. Value has been defaulted to '[java.util.*, kotlinx.android.synthetic.**]'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.403 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'indent_style' was found. Value has been defaulted to 'space'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.403 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - Value of '.editorconfig' property 'indent_size' is remapped from 'unset' to '-1'
17:49:48.404 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'indent_style' was found. Value has been defaulted to 'space'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.404 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - Value of '.editorconfig' property 'indent_size' is remapped from 'unset' to '-1'
17:49:48.405 [pool-1-thread-1] TRACE com.pinterest.ktlint.ruleset.standard.WrappingRule - 1: inserted newline after (
17:49:48.406 [pool-1-thread-1] TRACE com.pinterest.ktlint.ruleset.standard.WrappingRule - 1: inserted newline before )
17:49:48.406 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'indent_style' was found. Value has been defaulted to 'space'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.406 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - Value of '.editorconfig' property 'indent_size' is remapped from 'unset' to '-1'
17:49:48.407 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'ij_kotlin_allow_trailing_comma_on_call_site' was found. Value has been defaulted to 'false'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.409 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'ij_kotlin_allow_trailing_comma' was found. Value has been defaulted to 'false'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.410 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'indent_style' was found. Value has been defaulted to 'space'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.417 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - Value of '.editorconfig' property 'indent_size' is remapped from 'unset' to '-1'
17:49:48.418 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'indent_style' was found. Value has been defaulted to 'space'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.418 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - Value of '.editorconfig' property 'indent_size' is remapped from 'unset' to '-1'
17:49:48.420 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'ij_kotlin_packages_to_use_import_on_demand' was found. Value has been defaulted to '[java.util.*, kotlinx.android.synthetic.**]'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.420 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'indent_style' was found. Value has been defaulted to 'space'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.420 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - Value of '.editorconfig' property 'indent_size' is remapped from 'unset' to '-1'
17:49:48.421 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'indent_style' was found. Value has been defaulted to 'space'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.421 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - Value of '.editorconfig' property 'indent_size' is remapped from 'unset' to '-1'
17:49:48.421 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'indent_style' was found. Value has been defaulted to 'space'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.421 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - Value of '.editorconfig' property 'indent_size' is remapped from 'unset' to '-1'
17:49:48.421 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'ij_kotlin_allow_trailing_comma_on_call_site' was found. Value has been defaulted to 'false'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.422 [pool-1-thread-1] TRACE com.pinterest.ktlint.core.api.UsesEditorConfigProperties - No value of '.editorconfig' property 'ij_kotlin_allow_trailing_comma' was found. Value has been defaulted to 'false'. Setting the value explicitly in '.editorconfig' removes this message from the log.
17:49:48.424 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - 632ms / 1 file(s) / 2 error(s)

```
</details>

It looks like the WrappingRule is causing this behavior if I read the logs right. 🤔

**Comment by paul-dingemans** [comment]
Finally, I could reproduce the issue. The problem is caused by setting `indent_size` to value `unset`. This results in disabling the `Indent` rule.

What is your reasoning behind using `unset`? `unset` is not meant to fallback on the default value of ktlint (for this you would simply leave out the property entirely).

**Comment by ubuntudroid** [comment]
> What is your reasoning behind using `unset`?

That's actually a very good question... I found it somewhere (can't remember where) when trying to fix some issues I had with indentation much earlier.

I've removed it and yes, you are right! Everything works like a charm now. Going to close this one, thanks for bearing with me here! 🙏
